File: tests/cases/yield_from_test.py
@resumable
def generator1():
    for i in range(3):
        yield i


@resumable
def generator2():
    for j in range(3, 5):
        yield j


@resumable(lower_yield_from=True)
def yield_from():
    yield from generator1()
    yield from generator2()
    # print(map)


if __name__ == "__main__":
    arr = []
    for i in yield_from():
        arr.append(i)
    assert arr == [0, 1, 2, 3, 4]

File: tests/cases-py2many/yield_from_test.jl
using ResumableFunctions
@resumable function generator1()
    for i in (0:2)
        @yield i
    end
end

@resumable function generator2()
    for j in (3:4)
        @yield j
    end
end

@resumable function yield_from()
    for v in generator1()
        @yield v
    end
    for v in generator2()
        @yield v
    end
end

function main()
    arr = []
    for i in yield_from()
        push!(arr, i)
    end
    @assert(arr == [0, 1, 2, 3, 4])
end

main()
